azure-mgmt-graph-rbac/src/main/java/com/microsoft/azure/management/graphrbac/Application.java

```diff
@@ -0,0 +1,294 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.management.graphrbac;
+
+import com.microsoft.azure.management.apigeneration.Beta;
+import com.microsoft.azure.management.apigeneration.Fluent;
+import com.microsoft.azure.management.graphrbac.implementation.ApplicationInner;
+import com.microsoft.azure.management.resources.fluentcore.arm.models.HasId;
+import com.microsoft.azure.management.resources.fluentcore.arm.models.HasName;
+import com.microsoft.azure.management.resources.fluentcore.model.Appliable;
+import com.microsoft.azure.management.resources.fluentcore.model.Creatable;
+import com.microsoft.azure.management.resources.fluentcore.model.HasInner;
+import com.microsoft.azure.management.resources.fluentcore.model.Indexable;
+import com.microsoft.azure.management.resources.fluentcore.model.Updatable;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * An immutable client-side representation of an Azure AD application.
+ */
+@Fluent(ContainerName = "/Microsoft.Azure.Management.Fluent.Graph.RBAC")
+@Beta
+public interface Application extends
+        Indexable,
+        HasInner<ApplicationInner>,
+        HasId,
+        HasName,
+        Updatable<Application.Update> {
+    /**
+     * @return the object type
+     */
+    String objectType();
+
+    /**
+     * @return the application ID
+     */
+    String appId();
+
+    /**
+     * @return the application permissions
+     */
+    List<String> appPermissions();
+
+    /**
+     * @return whether the application is be available to other tenants
+     */
+    boolean availableToOtherTenants();
+
+    /**
+     * @return a collection of URIs for the application
+     */
+    List<String> identifierUris();
```

Comment:
does the order matter?  if not then should be a `Collection<String>`. But I suspect duplicates are also meaningless, so this could be `Set<String>` right?